lib/lease-manager.js

```diff
@@ -23,10 +24,30 @@ class LeaseManager extends EventEmitter {
 
   /**
    * Initialize the SecretProvider
+   * @returns {Promise}
    */
   initialize() {
-    this.provider.initialize((error, result) => {
-      this._onProviderInitialize(error, result);
+    return new Promise((resolve, reject) => {
+      this.provider.initialize().then((result) => {
+        this.status = STATUS.READY;
+        this.data = result.data;
+        this.lease_duration = result.lease_duration;
+        this.emit('ready');
+        this.error = null;
+        return resolve(this._renew());
+      }).catch((err) => {
+        this.status = STATUS.PENDING;
+        this.data = null;
+        this.lease_duration = 0;
+        if (this.listeners('error').length > 0) {
+          this.emit('error', err);
+        }
+        this.error = err;
+
+        // TODO: This isn't great. Ideally we should be rejecting error conditions and
+        // then handling any attempts at fixing the problem elsewhere.
+        return resolve(this.initialize());
```

Comment:
This is on the list to fix. See #43. I'm considering it out of scope for this issue but definitely something that needs to be fixed.